### DISCUSSION TITLE
Change Backlight instead of Luminance on LCD screens?

### DISCUSSION THREAD

**Discussion by torik42** [MonitorControl/MonitorControl#1382] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Thank you for the great App, it is (would be) much nicer to control a monitor via the OS than with the OSD.

My problem is, that changing the Brightness on a Fujitsu B27‑8 TS Pro does change what is called “Black level” in the OSD. Changing the contrast works as expected. Instead, I would like to change what is called “Brightness” in the OSD. I guess, it controls the backlight, which should also be accessible via DDC. The reason seems to be that MonitorControl changes “Luminance”:
https://github.com/MonitorControl/MonitorControl/blob/03355a0fa9f94d5789b7b266c4ef8dc1bcdbca53/MonitorControl/Enums/Command.swift#L171
Instead of what I guess is the backlight control:
https://github.com/MonitorControl/MonitorControl/blob/03355a0fa9f94d5789b7b266c4ef8dc1bcdbca53/MonitorControl/Enums/Command.swift#L110

I found [this source](https://www.poynton.ca/notes/brightness_and_contrast/index.html) explaining the three terms mainly for old CRT displays, but also mentioning LCD screens. The same issue exists when using other tools like ddcctl. And I haven’t found a way to test as I have not yet found a tool for mac, which easily lets me send other commands. That might even be a bad idea because some people have bricked their monitors by sending unsupported controls ([ddcctl issue](https://github.com/kfix/ddcctl/issues/89)). I also haven’t found a tool for mac, that allows to easily ask for the capabilities of the monitor ([ddcctl feature request](https://github.com/kfix/ddcctl/issues/58)).

I am not sure, whether also other users have this problem, or whether some monitors actually implement luminance as backlight luminance (see [above’s source again](https://www.poynton.ca/notes/brightness_and_contrast/index.html)). Then it would be great to have a switch between these options, or even a third control.

For now, I have the following questions:
1. Do you know a tool for macOS which lets me query the capabilities and send custom commands? ([This](https://github.com/itxch/ddcctl) fork of ddcctl has the ability to send, but not query the monitor to my understanding)
2. There is a VCP list in the advanced settings section for which I could not find any documentation. Does it allow sending custom commands?
3. If 2 is the case, does MonitorControl ask the monitor for supported commands and only send those?

### Expected behavior

MonitorControl to set the Monitors Brightness (probably backlight luminance) instead of black level.

### Anything else?

see above

### Environment Information (please complete the following information)

```markdown
- macOS version: 13.2.1 Ventura
- Mac model: MacBook Pro (16 inch, 2019)
- MonitorControl version: 4.1.0 Build 7034
- Monitor(s): Fujitsu B27‑8 TS Pro
- Monitor Cable(s)/Connection(s): USB‑C to DP cable
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment, marked as answer]
Hi @torik42 - you can use both MonitorControl and [BetterDisplay](https://betterdisplay.pro) to override the control code for Brightness/Luminance.

In BD, you can do it here:

<img width="942" alt="Screenshot 2023-05-31 at 16 30 28" src="https://github.com/MonitorControl/MonitorControl/assets/37590873/628a1f45-b330-4373-8266-22d6f9215a62">

In MC, you can do it here:

<img width="842" alt="Screenshot 2023-05-31 at 16 31 10" src="https://github.com/MonitorControl/MonitorControl/assets/37590873/a8d34ee9-0ca4-4780-b8eb-a432a524440b">

**Reply by torik42** [reply]
Thank you for the answer. That’s exactly the setting I was wondering about.

I guess, I just put a hex value there (`13` in my example). Is that correct? And does MC check, whether it is a supported command? (So that the risk of bricking the monitor is reduced.)

**Reply by waydabber** [reply]
No, it won't check, you have to take the risk. But bricking the display is not very high with an invalid DDC command (still can happen of course).

**Reply by torik42** [reply]
Okay, thanks. Works perfectly now.

For anyone else coming here, the correct setting for me was `Backlight Level: White` with code `6b`. (I can even hijack the volume control for black level/luminance.)
